docs/BR.md

```diff
@@ -2681,6 +2681,8 @@ Table: `GeneralName` within a `subjectAltName` extension
 | `iPAddress`                 | Y             | The entry MUST contain the IPv4 or IPv6 address that the CA has confirmed the Applicant controls or has been granted the right to use through a method specified in [Section 3.2.2.5](#3225-authentication-for-an-ip-address). The entry MUST NOT contain a Reserved IP Address. |
 | `registeredID`              | N             | -              |
 
+**Note**: The allowance for including P-Labels that do not conform to IDNA 2003 is an exception to best practice as defined in RFC 5280. These Requirements allow for the inclusion of P-Labels that do not conform with IDNA 2003 to support newer versions of the Unicode character repertoire, among other improvements to the various IDNA standards.
```

Comment:
@XolphinMartijn @CBonnell: Framing this as "an exception to best practice" seems wrong to me.  At IETF, "best practice" has evolved - see RFC8399, which updates RFC5280 and permits IDNA2008.
CABForum documents don't automatically recognise IETF document updates or obsoletions (and I think that's a feature, not a bug), but this doesn't mean we can never choose to recognise new IETF documents going forwards!
In this case, ISTM that the appropriate way forward is for the BRs to be updated to recognise RFC8399.  i.e., references to RFC5280 should become "RFC5280, as updated by RFC8399".

Reply:
Hi @robstradling, so if I'm understanding correctly, you're suggesting changing:

> The allowance for including P-Labels that do not conform to IDNA 2003 is an exception to best practice as defined in RFC 5280.  These Requirements allow for the inclusion of P-Labels that do not conform with IDNA 2003 to support newer versions of the Unicode character repertoire, among other improvements to the various IDNA standards.

to:

> The allowance for including P-Labels that do not conform to IDNA 2003 and IDNA 2008 is an exception to best practice as defined in RFC 5280 and RFC 8399.  These Requirements allow for the inclusion of P-Labels that do not conform with IDNA 2003 and IDNA 2008 to support Unicode Technical Standard #46 and its variants, among other changes to the various IDNA standards.

Reply:
Hi @CBonnell.  Actually I was trying to suggest that (1) the current best practice, as defined by "RFC 5280, as updated by RFC8399", is to permit P-Labels that conform to IDNA 2003 and/or IDNA 2008, and to reject all other Reserved LDH Labels; and that (2) the BRs should be updated to regard "RFC 5280, as updated by RFC8399" as best practice, rather than as an exception to best practice.
Does that make sense?

Reply:
I cribbed the "exception to best practice..." language from section 7.2.2:

> Note: The CA SHOULD update the revocation date in a CRL entry when it is determined that the
private key of the Certificate was compromised prior to the revocation date that is indicated in the
CRL entry for that Certificate. Backdating the revocationDate field is an exception to best practice
described in RFC 5280 (Section 5.3.2); however, these requirements specify the use of the
revocationDate field to support TLS implementations that process the revocationDate field as the
date when the Certificate is first considered to be compromised.

I think it's rather clear that the text is establishing the RFC(s) as best practice and some allowance within the BRs to be an exception to such best practice and not the other way around. But if you have improved concrete language that resolves the concern, perhaps that be added.

Reply:
Hi @CBonnell.  I think we're still talking past each other a bit.  Let me try to restate fully where I'm coming from...

Sectigo's view of the current situation is that:
- RFC5280 only permits IDNA 2003 to be used.
- The BRs currently require conformance to RFC5280.
- At IETF, RFC5280 has been updated by RFC8399, which adds permission for IDNA 2008 to be used.
- However, at CABForum, IETF document updates are not automatically incorporated into the BRs.
- The BRs were updated a while ago such that the LDH Label, Non-Reserved LDH Label, and XN-Label definitions now incorporate quotes from RFC5890, which is one of the documents that constitute IDNA 2008.  However, we don't believe that merely quoting some definitions from an RFC means that that entire RFC is assumed to be incorporated into the BRs, especially when that would conflict with other BR requirements (e.g., the requirement to conform to RFC5280, which only permits IDNA 2003).
- We can accept that the intention of the SC48v2 author/endorsers was to permit IDNA 2008, and so we feel that it's appropriate for a clean-up ballot to improve the language in line with that intent.

The gist of how I'd like to see the BRs updated:
- Explicitly permit IDNA 2008, by explicitly incorporating RFC8399 in the BRs.
- Allow the definition of "best practice" to move with the times, and not get stuck with updates to RFC5280 being labelled as "exceptions" to best practice.

I think the best way to achieve this is to update the BRs as follows:

1. Update the "P-Label" definition along these lines...
```
P‐Label: A XN‐Label that contains valid output of the Punycode algorithm (as defined in RFC 3492,
Section 6.3) from the fifth and subsequent positions.  These Requirements permit P-Labels that conform to RFC 5280 as updated by RFC 8399, meaning that both IDNA 2003 and IDNA 2008 may be used.
```

2. Add RFC 8399 to section 1.6.3 (References).

Then, when section 7.1.2 says "Except as explicitly noted, all normative requirements imposed by RFC 5280 shall apply, in addition to the normative requirements imposed by this document", this updated definition of P-Label would be instance of "Except as explicitly noted", and so the "normative requirements imposed by RFC 5280" that only permit IDNA 2003 would be overriden.

Reply:
Hi @robstradling, I believe that the current language is sufficiently clear from a compliance standpoint that IDNA 2008 (and others) are allowed (the definition of P-Label and associated requirements are quite explicit on this point), but I agree that the language can be improved to aid readers.

> P‐Label: ...  These Requirements permit P-Labels that conform to RFC 5280 as updated by RFC 8399, meaning that both IDNA 2003 and IDNA 2008 may be used.

This language is problematic, as it is now unclear whether XN-labels that conform with UTS#46 (or other variants) but not IDNA 2008 are allowed, whereas it is clear that they are allowed in the current language. This, coupled with RFC 8399's SHOULD-level requirement for adherence to IDNA 2008, especially muddies the waters.

This is why I suggested the language in my previous post: it gives a nod to RFC 5280 and 8399 but does not establish them as normative in this area, because doing so adds confusion. I think the clearest course is to clearly establish what is allowed directly in the BRs and add informative references to the RFCs.

Reply:
Hi @CBonnell.  I still don't agree that the current language explicitly permits IDNA 2008 (and others), but it looks like we'll have to agree to disagree on that point.
TBH, I'm not familiar with IDNs that are neither IDNA 2003 nor IDNA 2008 but are instead "UTF#46 (or other variants)", so I can't comment on whether or not the BRs should permit them.  You make a good point that "RFC 8399's SHOULD-level requirement for adherence to IDNA 2008" would impact whether or not they're permitted. 
 I suppose it would also impact IDNA 2003 IDNs that don't conform to IDNA 2008.
So I understand now that you don't consider post-RFC5280 IETF "best practice" to be a suitable baseline for the BRs to use, which is why you want to frame the language as "an exception to best practice as defined in RFC 5280".  I guess I could live with that.
I do agree though that "the clearest course is to clearly establish what is allowed directly in the BRs and add informative references to the RFCs".

Reply:
```suggestion
**Note**: As an explicit exception from RFC 5280, P-Labels are permitted to not conform to IDNA 2003. These Requirements allow for the inclusion of P-Labels that do not conform with IDNA 2003 to support newer versions of the Unicode character repertoire, among other improvements to the various IDNA standards.
```